core-web/libs/sdk/angular/next/components/dotcms-layout-body/components/contentlet/contentlet.component.ts

```diff
@@ -0,0 +1,116 @@
+import { DotCMSContentlet, DotContentletAttributes } from '@dotcms/uve/types';
+
+import { DynamicComponentEntity } from '../../../../models';
+import { DotCMSContextService } from '../../../../services/dotcms-context/dotcms-context.service';
+import { FallbackComponent } from '../fallback-component/fallback-component.component';
+
+/**
+ * Contentlet component that renders DotCMS content with development mode support
+ *
+ * @component
+ * @param {DotCMSContentlet} contentlet - The contentlet to be rendered
+ * @param {string} container - The container identifier
+ */
+@Component({
+    selector: 'dotcms-contentlet',
+    standalone: true,
+    imports: [FallbackComponent, AsyncPipe, NgComponentOutlet],
+    template: `
+        @if ($UserComponent()) {
+            <ng-container
+                *ngComponentOutlet="
+                    $UserComponent() | async;
+                    inputs: { contentlet: $contentlet() ?? contentlet }
+                " />
+        } @else if ($UserNoComponent() && $isDevMode()) {
+            <dotcms-fallback-component
+                [UserNoComponent]="$UserNoComponent()"
+                [contentlet]="$contentlet() ?? contentlet" />
+        }
+    `,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ContentletComponent implements OnChanges {
+    @Input({ required: true }) contentlet!: DotCMSContentlet;
+    @Input({ required: true }) container!: string;
+    @ViewChild('contentletRef') contentletRef!: ElementRef;
+    @HostBinding('attr.data-dot-object') dotObject = 'contentlet';
+
+    #dotcmsContextService = inject(DotCMSContextService);
+
+    $contentlet = signal<DotCMSContentlet | null>(null);
+    $UserComponent = signal<DynamicComponentEntity | null>(null);
+    $UserNoComponent = signal<DynamicComponentEntity | null>(null);
+    $isDevMode = signal(false);
+    $haveContent = signal(false);
+    $style = computed(() =>
+        this.$isDevMode() && this.$haveContent() ? { minHeight: '4rem' } : {}
+    );
+    $dotAttributes = computed<DotContentletAttributes>(() => {
+        const contentlet = this.$contentlet();
+        if (!contentlet || !this.$isDevMode()) return {} as DotContentletAttributes;
+
+        return getDotContentletAttributes(contentlet, this.container);
+    });
+
+    @HostBinding('attr.data-dot-identifier') identifier: string | null = null;
+    @HostBinding('attr.data-dot-basetype') basetype: string | null = null;
+    @HostBinding('attr.data-dot-title') title: string | null = null;
+    @HostBinding('attr.data-dot-inode') inode: string | null = null;
+    @HostBinding('attr.data-dot-type') type: string | null = null;
+    @HostBinding('attr.data-dot-container') containerAttribute: string | null = null;
+    @HostBinding('attr.data-dot-on-number-of-pages') onNumberOfPages: string | null = null;
+    @HostBinding('style') styleAttribute: { [key: string]: unknown } | null = null;
```

Comment:
This might be a good candidate for a directive, following Angular documentation:

- https://angular.dev/api/core/HostBinding
- https://angular.dev/guide/directives/directive-composition-api#including-inputs-and-outputs

Reply:
Indeed, this can be a Directive, but I'm not sure if we will be using this outside of this component 🤔

Reply:
It's for readability more than anything else, but we can skip this

Reply:
Alright, let's skip it for now